app/code/Magento/Inventory/Model/Source/Validator/PreventDisableValidator.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Inventory\Model\Source\Validator;
```

Comment:
We need to move this validator to `InventoryCatalog`
Because `DefaultSource` is the concept of  `InventoryCatalog` but not  `Inventory`

Reply:
Sure, I'll do that. Thanks